src/core/profiling/endoscope_backend.h

```diff
@@ -0,0 +1,163 @@
+#include <grpc/support/sync.h>
+#include <grpc/support/time.h>
+#include <stdlib.h>
+
+#ifndef ENDOSCOPE_BACKEND_H_
+#define ENDOSCOPE_BACKEND_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Data types */
+
+#define ENDO_MUTEX_TYPE gpr_mu
+#define ENDO_INT64 gpr_int64
+#define ENDO_INDEX unsigned short
+#define ENDO_EMPTY (ENDO_INDEX)(-1)
+
+/* Public Macros */
+
+#define ENDOSCOPE_INIT(base) endoscope_init(base)
+#define ENDOSCOPE_DESTROY(base) endoscope_destroy(base)
+#define ENDOSCOPE_BEGIN(base, name) \
+        do { \
+          ENDO_INT64* temp_ = endoscope_begin(base, name, \
+              __FILE__, __LINE__, __PRETTY_FUNCTION__); \
+          *temp_ = ENDO_CYCLENOW(); \
+        } while (0)
+#define ENDOSCOPE_END(base, name) endoscope_end(base, name, ENDO_CYCLENOW())
+#define ENDOSCOPE_EVENT(base, name) endoscope_event(base, name, \
+            __FILE__, __LINE__, __PRETTY_FUNCTION__, ENDO_CYCLENOW())
+#define ENDOSCOPE_ERROR(base, name) endoscope_error(base, name, \
+            __FILE__, __LINE__, __PRETTY_FUNCTION__, ENDO_CYCLENOW())
+
+/* Private macros */
+
+#define ENDO_CYCLENOW() endoscope_cyclenow()
+#define ENDO_SYNCCLOCK(cycle, time) endoscope_syncclock(cycle, time)
+#define ENDO_GETTID() endoscope_gettid()
+#define ENDO_MUTEX_INIT(mu) \
+        do { \
+          mu = (void *)malloc(sizeof(ENDO_MUTEX_TYPE)); \
+          gpr_mu_init((ENDO_MUTEX_TYPE*)(mu)); \
+        } while (0)
+#define ENDO_MUTEX_DESTROY(mu) \
+        do { \
+          gpr_mu_destroy((ENDO_MUTEX_TYPE*)(mu)); \
+          free(mu); \
+        } while (0)
+#define ENDO_LOCK(mu) gpr_mu_lock((ENDO_MUTEX_TYPE*)(mu))
+#define ENDO_UNLOCK(mu) gpr_mu_unlock((ENDO_MUTEX_TYPE*)(mu))
+
+/* Constant Parameters (max capacity 65534) */
+
+#define hash_size 97
```

Comment:
They are previously const int in C++, but C does not support const int as array dimension, so I just replaced const int with #define. Since they are used for constant array dimension, they need to be in the header, although don't need to be public. I am changing the names.